server/api/controllers/liveSermon/get-live-sermon.js

```diff
@@ -28,16 +30,36 @@ module.exports = {
         let data = await LiveSermon.find({
           _id: sermonId,
           isDeleted: false,
-          isPublished
+          isPublished,
         }).populateAll();
         if (data.length === 0) {
           return exits.success([]);
         }
         return exits.success(data);
       }
 
-      let data = await LiveSermon.find({ isDeleted: false, isPublished }).sort('updatedAt DESC').populateAll();
-      sails.log.info('Retrieving live sermons');
+      let data = await LiveSermon.find({ isDeleted: false, isPublished })
```

Comment:
i would make a couple changes here.
1. remove `isPublished` as API's input params
2. If a `sermonId` passed in input, simply find and return.
3. If `sermonId` not passed as input, find the most recently updated sermon that has `startTime < endTime && now > startTime && now < endTime`

Reply:
you will have to also search up the whole codebase for pages that calls this API and update them accordingly